src/sources/util/encoding_config.rs

```diff
@@ -0,0 +1,8 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct EncodingConfig {
+    #[serde(default)]
+    pub charset: Option<&'static encoding_rs::Encoding>,
```

Comment:
In the current form of `EncodingConfig`, it's not strictly needed (since it's the only item here and using the struct implies specifying a value to `charset`) but if we place the source-level encoding configuration as `codec` here (eg: from the http source), `charset` would need to be optional, since it does not need to take a value always. I was also trying to keep this in sync with the [sink encoding configuration](https://github.com/timberio/vector/blob/3a48a0a47dd1e1b59113041a96cdced29937cc19/src/sinks/util/encoding/config.rs#L23-L34) -- when `charset` support is added there, it will be as an `Option`.

If you would prefer to turn this into an `Option` later when the struct gets other items, I can remove it from here for now.

Reply:
> If you would prefer to turn this into an Option later when the struct gets other items, I can remove it from here for now.

Yep, we should prefer this, a bit of complexity less for now.

Reply:
Fair enough, removed the `Option` here as discussed.